port/unix/omrsysinfo.c

```diff
@@ -7687,8 +7687,104 @@ omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallb
 	portLibrary->mem_free_memory(portLibrary, command);
 	closedir(dir);
 	return callback_result;
-#else /* defined(LINUX) */
+#elif defined(AIXPPC) /* defined(LINUX) */
+#define MAX_PROCS 256
+#define INITIAL_ARGS_SIZE 8192
+	pid_t index = 0;
+	int32_t numProcs = 0;
+	uintptr_t callback_result = 0;
+	struct procentry64 *procs = (struct procentry64 *)portLibrary->mem_allocate_memory(
+			portLibrary,
+			MAX_PROCS * sizeof(struct procentry64),
+			OMR_GET_CALLSITE(),
+			OMRMEM_CATEGORY_PORT_LIBRARY);
+	if (NULL == procs) {
+		return OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED;
+	}
+	for (;;) {
+		numProcs = getprocs64(procs, sizeof(struct procentry64), NULL, 0, &index, MAX_PROCS);
+		if (-1 == numProcs) {
+			int32_t rc = findError(errno);
+			portLibrary->error_set_last_error(portLibrary, errno, rc);
+			Trc_PRT_failed_to_getprocs64(rc);
+			portLibrary->mem_free_memory(portLibrary, procs);
+			return (uintptr_t)(intptr_t)rc;
+		}
+		if (0 == numProcs) {
+			break;
+		}
+		for (int32_t i = 0; i < numProcs; i++) {
```

Comment:
Please use proper C syntax; declarations are not permitted in `for` statements, nor after general statements in a block.